add new tutorial: Assembly metagenomics (#3751)
galaxyproject/training-material: topics/metagenomics/tutorials/metagenomics-assembly/tutorial.md

---
layout: tutorial_hands_on

title: "Metagenomic data assembly"

> zenodo_link: ""

Reviewer: ```suggestion
zenodo_link: "https://zenodo.org/record/6958893"
```

questions:
  - "Why metagenomic data should be assembled?"
  - "What is the difference between co-assembly and individual assembly?"
  - "What is the difference between reads, contigs and scaffolds?"
  - "How tools based on De Bruijn graph work?"  
  - "How to assess the quality of metagenomic data assembly?"
objectives:
  - "Describe what an assembly is"
  - "Describe what de-replication is"
  - "Explain the difference between co-assembly and individual assembly"
  - "Explain the difference between reads, contigs and scaffolds"
  - "Explain how tools based on De Bruijn graph work"
  - "Apply appropriate tools for analyzing the quality of metagenomic data"
  - "Construct and apply simple assembly pipelines on short read data"
  - "Apply appropriate tools for analyzing the quality of metagenomic assembly"
  - "Evaluate the Quality of the Assembly with Quast, Bowtie2, and CoverM-Genome"
time_estimation: "2H"
key_points:
  - "Assembly groups reads into contigs and scafolds."
  - "De Brujin Graphs use k-mers to assembly reads"
  - "MetaSPAdes and Megahit are assemblers"
  - "Quast is the tool to assess the assembly quality"
contributors:

>   - PlushZ

Reviewer: Please also add yourself to the `CONTRIBUTORS.yaml` file

Author: I see myself there already, did I misunderstand?

For an in-depth analysis of the structure and functions of the coffee microbiome, a temporal shotgun metagenomic study (six time points) was performed.

Real datasets are available for download in [ENA](https://www.ebi.ac.uk/ena/browser/view/PRJEB24129?show=reads), as well as metadata about samples. There are six samples of a coffee fermentation system sequenced with Illumina MiSeq utilizing whole genome sequencing. This dataset contains paired-end reads.

#### Upload dataset to Galaxy
First we need to get some data into our history. There are different ways to upload data into the Galaxy. You can upload files from your computer, or Galaxy can also fetch data directly from external sources.

In case of a not very large dataset it’s more convenient to upload data directly from your computer to Galaxy. In case of a large dataset, we can use FTP server or the [Galaxy Rule-based Uploader]({% link topics/galaxy-interface/tutorials/upload-rules/tutorial.md %}).

> ### {% icon hands_on %} Hands-on: Upload data into Galaxy
>
> 1. Create new history
>
>    {% snippet faqs/galaxy/histories_create_new.md %}
>
> 2. At the top of the Tools panel (on the left), click “Upload Data”

> > ![Upload data](../../images/metagenomics-assembly/upload-data.png){:width="40%"}

Reviewer: To avoid duplication, please use the "import via url" snippet and remove the screenshots

`{% snippet  faqs/galaxy/datasets_import_via_link.md %}`

There are three challenges to metagenomics assembly: 1) differences in genome coverage between samples, resulting from differences in abundance, 2) the fact that different species often share conserved regions ({%cite kececioglu2001%}), and 3) the presence of multiple strains of a single species ({%cite miller2010%}). Since SPAdes handles non-uniform coverage, it is useful for assembling simple communities, but metaSPAdes also handles other problems, allowing it to assemble complex communities' metagenomes.

#### Co-assembly

> ### {% icon hands_on %} Hands-on: Assembly with MetaSPAdes
>
> > ### {% icon comment %} Comment
> >
> > In the next steps we use **metaSPAdes** assembler. **MetaSPAdes** is a versatile metagenomic assembler ({%cite nurk2017%}). As input for metaspades it can accept short reads. However, there is an option to use additionally long reads besides short reads to produce hybrid input.
> {: .comment} 
>
> 1. Name a new history with the name “Assembly with MetaSPAdes”
>

> > 2. Run **MetaSPAdes**{% icon tool %} with parameters:

Reviewer 2: Can you add for all tools their link like explained: https://training.galaxyproject.org/training-material/topics/contributing/tutorials/create-new-tutorial-content/tutorial.html#tool-links? Thanks

>     - {% icon param-file %} *"Pair-end reads input format"*: paired_interlaced
>     - {% icon param-file %} *"FASTQ file(s): interlaced"*: collection of short reads
>     - {% icon param-file %} *"Select k-mer detection option"*: manual
>     - {% icon param-file %} *"K-mer size values"*: 21,33,55,77
> 3. Rename output to “Output - assembly with MetaSPAdes”
{: .hands_on}

> ### {% icon details %} Output of MetaSPAdes
>
> **MetaSPAdes** outputs contigs and scaffolds.

> Scaffolds are segments of genome sequence reconstructed from end-sequenced whole-genome shotgun clones. Contigs and gaps make up scaffolds. Contigs are contiguous lengths of genomic sequences in which bases are known to a high degree of certainty. The gaps occur when reads from the two sequenced ends of at least one fragment overlap with other reads from two different contigs (as long as the arrangement is otherwise consistent with the contigs being adjacent). It is possible to estimate the number of bases between contigs based on fragment lengths.

Reviewer: each line in a box needs to start with `>` or it won't be displayed properly

```suggestion
> Scaffolds are segments of genome sequence reconstructed from end-sequenced whole-genome shotgun clones. Contigs and gaps make up scaffolds. Contigs are contiguous lengths of genomic sequences in which bases are known to a high degree of certainty. The gaps occur when reads from the two sequenced ends of at least one fragment overlap with other reads from two different contigs (as long as the arrangement is otherwise consistent with the contigs being adjacent). It is possible to estimate the number of bases between contigs based on fragment lengths.
```

>
{: .details}

> ### {% icon comment %} Co-assembly
>
> We executed a so-called co-assembly. Let's learn a bit more about co-assembly. Basically, we performed the assembly on the bunch of samples taking reads from all samples simultaneously while assembling.
>

> > ![Co-assembly](../../images/metagenomics-assembly/co-assembly.png){:width="60%"}

Reviewer: the alt text (between the `[]`) should describe the image in detail. imagine you have to describe what is on the image to somebody who cannot see it, it can be as long as needed. Could you update the alt text of all the images?

>
> ![Co-assembly](../../images/metagenomics-assembly/co-assembly-process.png){:width="80%"}
>
> In these cases, co-assembly is reasonable:
> - Same sample
> - Same sampling event
> - Longitudinal sampling of the same site
> - Related samples 
>
> Sometimes it is important to run tools both on individual samples and on all pooled samples.
> It is useful to use both approaches: individual assembly and co-assembly and benchmark them in order to get the intuition of more applicable techniques for the certain dataset. In case of individual assembly we have the possibility to de-replicate the genome after binning. Co-assembly is more commonly used than individual assembly and then de-replication after binning. Nevertheless, the de-replication approach should be taken into account while choosing the assembly method.
{: .comment}

##### Assessment of assembly

To determine the number of reads used to build an assembly, we want to know the percentage of reads that were used.

###### Extraction of % reads used in assemblies
We map input reads onto assemblies in order to extract this information.

> In order to calculate the percentage of reads used for assembly, we use **Bowtie2** to map raw short-read data to the assembly we got using **MetaSPAdes**.

Reviewer 2: Please add Bowtie2 citation. Thanks

Assemblies can be evaluated with **metaQUAST** ({%cite mikheenko2016%}), metagenomics mode of **QUAST** ({%cite gurevich2013%}).

> ### {% icon hands_on %} Hands-on: General metrics with metaQUAST
>
> 1. Run **Quast**{% icon tool %} with parameters:
>     - {% icon param-file %} *"Contigs/scaffolds file"*: outputs of the assemblies
>     - {% icon param-file %} *"Type of assembly"*: metagenome
>     - {% icon param-file %} *"Output files"*: history
>           - HTML report 
>           - PDF report 
>           - Tabular reports 
>           - Log file 
> 
{: .hands_on}

##### Aggregate all metrics

To complement already informative **metaQuast** reports, we will show you a stunning way to create a combined report.
Using **MultiQC** ({%cite ewels2016%}), one can generate a nice report with all metrics combined. When you want to combine results and display them visually with different graphs, this tool is a good choice. Reports from **MultiQC** are combined into a webpage.

> ### {% icon hands_on %} Hands-on: Aggregation of all metrics
>
> 1. Run **MultiQC**{% icon tool %} with parameters:

> >     - {% icon param-file %} *"Which tool used to generate report"*: Quast
> >     - {% icon param-file %} *"Data"*: all outputs of Quast
> > 2. Add report for **Bowtie 2**
> >     - {% icon param-file %} *"Which tool used to generate report"*: Bowtie2
> >     - {% icon param-file %} *"Data"*: all outputs of Bowtie2
> > 

Reviewer 2: ```suggestion
>    - In *"Results"*:
>        - {% icon param-repeat %} *"Insert Results"*
>            - *"Which tool was used generate logs?"*: `Quast`
>                -  {% icon param-file %} *"Data"*: all outputs of Quast
>        - {% icon param-repeat %} *"Insert Results"*
>            - "Which tool used to generate report"*: Bowtie2
>                - {% icon param-file %} *"Data"*: all outputs of Bowtie2
> 
```

{: .hands_on}

#### Individual assembly
So far we performed co-assembly. For the assembly techniques with the de-replication involved, it is important to know how to do individual assembly. A corresponding hands-on section with steps of individual assembly of short reads samples is below.
We will use a Megahit assembler ({%cite li2015%}) for individual assembly. This is fortunate that Megahit can perform both options: individual and co-assembly.

> > ### {% icon details %} De-replication

Reviewer 2: Dereplication should be moved later

> We will perform steps from 1 to 3 in this tutorial a bit later while steps 4 - 8 will be considered in the following tutorial - Binning tutorial.
{: .details}

> ### {% icon hands_on %} Hands-on: Individual assembly of short-reads samples with **MEGAHIT**
> 1. Create new history
>
>    {% snippet faqs/galaxy/histories_create_new.md %}
>
> 2. Name a new history "Individual assembly of short-reads samples with MEGAHIT"
> 3.  Run **MEGAHIT**{% icon tool %} with parameters:
>     - {% icon param-file %} *"Select your input option"*: paired-end collection
>     - {% icon param-file %} *"Run in batch mode?"*: Run individually
>     - {% icon param-file %} *"Select a paired collection"*: dataset collection
>     - {% icon param-file %} *"K-mer specification method"*: klim_method
>     - {% icon param-file %} *"Minimum kmer size"*: 21
>     - {% icon param-file %} *"Maximum kmer size"*: 91
>     - {% icon param-file %} *"Increment of kmer size of each iteration"*: 12
>
> > ### {% icon comment %} Comment
> >
> > **MEGAHIT** produced a collection of output assemblies - one per sample - that can be proceeded further in binning step and then de-replication.
> {: .comment} 
>
{: .hands_on}

> Well done! {% icon trophy %}

Reviewer 2: Conclusion is missing
